File: feral-nodes/android-app/src/main/java/ai/feral/node/FeralV2Tokens.kt
package ai.feral.node

import androidx.compose.animation.core.CubicBezierEasing
import androidx.compose.animation.core.tween
import androidx.compose.ui.graphics.Color
import androidx.compose.ui.unit.dp
import androidx.compose.ui.unit.sp

/**
 * Feral v2 design tokens — canonical source of truth for the Android
 * ambient-OS UI. Mirrors `feral-client-v2/src/styles/tokens.css` and
 * `ios-app/App/FeralV2Tokens.swift`. Keep all three in lock-step.
 *
 * Persona-continuity rule: Orb / Chat / Voice screens must consume these
 * tokens — no hardcoded Color, FontSize, or Dp literals.
 */
object FeralV2Tokens {
    // ── Base + surfaces ─────────────────────────────────────────
    val bgBase      = Color(0xFF0B0B0D)
    val bgDeep      = Color(0xFF060607)
    val surface0    = Color(0x0FFFFFFF)
    val surface1    = Color(0x17FFFFFF)
    val surface2    = Color(0x21FFFFFF)
    val surfaceElev = Color(0x2EFFFFFF)

    val hairline        = Color(0x14FFFFFF)
    val hairlineStrong  = Color(0x24FFFFFF)
    val hairlineFocus   = Color(0x38FFFFFF)

    // ── Text ────────────────────────────────────────────────────
    val textPrimary    = Color(0xFFF5F5F7)
    val textSecondary  = Color(0xFFA1A1A8)
    val textTertiary   = Color(0xFF6E6E76)
    val textInverse    = Color(0xFF0B0B0D)

    // ── Accent (interactive only) ───────────────────────────────
    val accent     = Color(0xFF0A84FF)
    val accentSoft = Color(0x2E0A84FF)
    val accentRing = Color(0x590A84FF)

    // ── Semantic state ──────────────────────────────────────────
    val stateLive  = Color(0xFF30D158)
    val stateWarn  = Color(0xFFFFD60A)
    val stateError = Color(0xFFFF453A)

    // ── Radii ───────────────────────────────────────────────────
    val radiusXS = 6.dp
    val radiusSM = 10.dp
    val radiusMD = 14.dp
    val radiusLG = 20.dp

    // ── Type scale ──────────────────────────────────────────────
    val sizeXS   = 11.sp
    val sizeSM   = 12.sp
    val sizeBase = 13.sp
    val sizeMD   = 15.sp
    val sizeLG   = 18.sp
    val sizeXL   = 22.sp
    val size2XL  = 32.sp

    // ── Motion ──────────────────────────────────────────────────
    const val DUR_FAST_MS = 120
    const val DUR_BASE_MS = 180
    const val DUR_SLOW_MS = 320
    val easeOut = CubicBezierEasing(0.16f, 1f, 0.3f, 1f)
    fun baseAnim() = tween<Float>(durationMillis = DUR_BASE_MS, easing = easeOut)
}

File: feral-nodes/ios-app/App/FeralV2Tokens.swift
//
//  FeralV2Tokens.swift
//  Feral iOS — shared design tokens for the v2 ambient-OS UI.
//
//  Canonical source of truth for v2 colors, type, radii, and motion on
//  iOS. Mirrors `feral-client-v2/src/styles/tokens.css`. Keep both files
//  in lock-step; drift means the mobile surface looks different from the
//  web client.
//
//  Persona-continuity rule: Orb / Chat / Voice screens must read from
//  these tokens only — no hardcoded SwiftUI Colors.
//
import SwiftUI

public enum FeralV2Tokens {
    // MARK: - Base + surfaces
    public static let bgBase       = Color(red: 0.043, green: 0.043, blue: 0.051)   // #0B0B0D
    public static let bgDeep       = Color(red: 0.024, green: 0.024, blue: 0.027)   // #060607
    public static let surface0     = Color.white.opacity(0.06)
    public static let surface1     = Color.white.opacity(0.09)
    public static let surface2     = Color.white.opacity(0.13)
    public static let surfaceElev  = Color.white.opacity(0.18)

    public static let hairline        = Color.white.opacity(0.08)
    public static let hairlineStrong  = Color.white.opacity(0.14)
    public static let hairlineFocus   = Color.white.opacity(0.22)

    // MARK: - Text
    public static let textPrimary    = Color(red: 0.96, green: 0.96, blue: 0.97)
    public static let textSecondary  = Color(red: 0.63, green: 0.63, blue: 0.66)
    public static let textTertiary   = Color(red: 0.43, green: 0.43, blue: 0.46)
    public static let textInverse    = Color(red: 0.043, green: 0.043, blue: 0.051)

    // MARK: - Accent (interactive-only)
    public static let accent       = Color(red: 0.039, green: 0.518, blue: 1.0)    // #0A84FF
    public static let accentSoft   = Color(red: 0.039, green: 0.518, blue: 1.0).opacity(0.18)
    public static let accentRing   = Color(red: 0.039, green: 0.518, blue: 1.0).opacity(0.35)

    // MARK: - Semantic state (never decorative)
    public static let stateLive   = Color(red: 0.188, green: 0.820, blue: 0.345)   // #30D158
    public static let stateWarn   = Color(red: 1.000, green: 0.839, blue: 0.039)   // #FFD60A
    public static let stateError  = Color(red: 1.000, green: 0.271, blue: 0.227)   // #FF453A

    // MARK: - Radii
    public static let radiusXS: CGFloat = 6
    public static let radiusSM: CGFloat = 10
    public static let radiusMD: CGFloat = 14
    public static let radiusLG: CGFloat = 20

    // MARK: - Type scale
    public static let sizeXS: CGFloat   = 11
    public static let sizeSM: CGFloat   = 12
    public static let sizeBase: CGFloat = 13
    public static let sizeMD: CGFloat   = 15
    public static let sizeLG: CGFloat   = 18
    public static let sizeXL: CGFloat   = 22
    public static let size2XL: CGFloat  = 32

    // MARK: - Motion
    public static let durFast: Double = 0.12
    public static let durBase: Double = 0.18
    public static let durSlow: Double = 0.32
    public static let easeOut = Animation.timingCurve(0.16, 1, 0.3, 1, duration: durBase)
}
